### ISSUE TITLE
SKLearnClassifier not working properly

### ISSUE THREAD

**Issue by J-ATJ** [keras-team/keras#20884]
Some day ago I was trying to add a Keras model to scikit-learn Pipeline using SKLearnClassifier (wrapper), but I got this error when doing **.fit()** (notwithstanding the model was already compiled):

_**RuntimeError**: Given model needs to be compiled, and have a loss and an optimizer_.

So I made this example code to check if the **clone_model** function that the API uses does indeed keep the compiled model:

```python
from keras.layers import Dense, Input
from keras.models import Sequential, clone_model
clf = Sequential()
clf.add(Input((7,)))
clf.add(Dense(8, activation="relu"))
clf.add(Dense(1, activation="sigmoid"))
clf.compile(optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"])
print("Original compiled?", clf.compiled)
cloned = clone_model(clf)
print("Cloned compiled?", cloned.compiled)
```

And the output:
```cmd
Original compiled? True
Cloned compiled? False
```

I think that's probably the reason why SKLearnClassifier is not working well.
Unfortunately, I couldn't find a way to solve this issue.

**Comment by de-odex** [comment]
I was able to work around this issue by passing a function to `SKLearnClassifier` instead of the instance. Not sure why this works though
